guides/real_time/presence.md

```diff
@@ -122,6 +124,120 @@ Bob (count: 1)
+      msg = {__MODULE__, {:join, user_data}}
+      Phoenix.PubSub.local_broadcast(Hello.PubSub, "proxy:#{topic}", msg)
+    end
+
+    for {user_id, presence} <- leaves do
+      metas =
+        case Map.fetch(presences, user_id) do
+          {:ok, presence_metas} -> presence_metas
+          :error -> []
+        end
+
+      user_data = %{id: user_id, user: presence.user, metas: metas}
+      msg = {__MODULE__, {:leave, user_data}}
+      Phoenix.PubSub.local_broadcast(Hello.PubSub, "proxy:#{topic}", msg)
+    end
+
+    {:ok, state}
+  end
+```
+
+You can see that we are broadcasting events for the joins and leaves. These will be listened to by the LiveView process. You'll also see that we use "proxy" channel when broadcasting the joins and leaves. This is because we don't want our LiveView process to receive the presence events directly. We can add a few helper functions so that this particular implementation detail is abstracted from the LiveView module.
+
+```elixir
+  def list_online_users(), do: list("online_users") |> Enum.map(fn {_id, presence} -> presence end)
+
+  def track_user(name, params), do: track(self(), "online_users", name, params)
+
+  def subscribe(), do: Phoenix.PubSub.subscribe(Hello.PubSub, "proxy:online_users")
+```
+
+Now that we have our presence module set up and broadcasting events, we can create a LiveView. Create a new file `lib/hello_web/live/online/index.ex` with the following contents:
+
+```elixir
+defmodule HelloWeb.OnlineLive do
+  use HelloWeb, :live_view
+
+  def mount(params, _session, socket) do
+    socket = stream(socket, :presences, [])
+    socket =
+    if connected?(socket) do
+      HelloWeb.Presence.track_user(params["name"], %{id: params["name"]})
+      HelloWeb.Presence.subscribe()
+      stream(socket, :presences, HelloWeb.Presence.list_online_users())
+    else
+       socket
+    end
+
+    {:ok, socket}
+  end
+
+  def render(assigns) do
+    ~H"""
+    <ul id="online_users" phx-update="stream">
+      <li :for={{dom_id, %{id: id, metas: metas}} <- @streams.presences} id={dom_id}><%= id %> (<%= length(metas) %>)</li>
+    </ul>
+    """
+  end
+
+  def handle_info({HelloWeb.Presence, {:join, presence}}, socket) do
+    {:noreply, stream_insert(socket, :presences, presence)}
+  end
+
+  def handle_info({HelloWeb.Presence, {:leave, presence}}, socket) do
+    if presence.metas == [] do
+      {:noreply, stream_delete(socket, :presences, presence)}
+    else
+      {:noreply, stream_insert(socket, :presences, presence)}
+    end
+  end
+end
+```
+
+If we add this route to the `lib/hello_web/router.ex`:
+
+```elixir
+    live "/online/:name", OnlineLive, :index
+```
+
+Then we can navigate to http://localhost:4000/online/Alice in one tab, and http://localhost:4000/online/Bob in another, you'll see that the presences are tracked, along with the number of presences per user. Opening and closing tabs with various users will update the presence list in real-time.
+
 ## Making it safe
```

Comment:
Should we make this a subsection of the channel one?

Reply:
This is the channels one? Do you mean should we make the content match and just change the URLs?

https://github.com/phoenixframework/phoenix/blob/d33f7bd034d1ac34145ea1a680bb79e9e2ae0d4f/guides/real_time/presence.md?plain=1#L118